pkg/backup/backup.go

```diff
@@ -154,6 +171,15 @@ type backupContext struct {
 	// resource, from either the networking.k8s.io or extensions api groups. We only want to back them
 	// up once, from whichever api group we see first.
 	networkPoliciesBackedUp bool
+
+	// backedUpItems keeps track of items that have been backed up already.
+	backedUpItems map[itemKey]struct{}
+
+	// additionalItems is a list of items returned by Actions that should be additionally backed up.
+	additionalItems []itemKey
+
+	// resources is a map of GroupResource->APIResource that we've seen so far.
```

Comment:
What would you think about adding extra functionality to our discovery helper so that we could more easily get a client from a GVR or GVK? In other words, when we refresh discovery, build up maps that lets us find an APIResource given either a GroupResource or a GroupKind? That would make it much easier to get a client, and we wouldn't have to rely on either iterating through or keeping track of what we've seen.

Reply:
Yeah, that's probably a better place to put the logic. I'll make the switch